src/Uno.UWP/Devices/Sensors/Pedometer.cs

```diff
@@ -0,0 +1,67 @@
+#if __IOS__ || __ANDROID__
+using System;
+using System.Threading.Tasks;
+using Windows.Foundation;
+
+namespace Windows.Devices.Sensors
+{
+	public partial class Pedometer
+	{
+		private readonly static object _syncLock = new object();
+		
+		private static Pedometer _instance;
+		private static bool _initializationAttempted;
+
+		private TypedEventHandler<Pedometer, PedometerReadingChangedEventArgs> _readingChanged;
+		
+		public static IAsyncOperation<Pedometer> GetDefaultAsync()
+		{
+			if (_initializationAttempted)
+			{
+				return Task.FromResult(_instance).AsAsyncOperation();
+			}
+			lock (_syncLock)
+			{
+				if (!_initializationAttempted)
+				{
+					_instance = TryCreateInstance();
+					_initializationAttempted = true;
+				}
+				return Task.FromResult(_instance).AsAsyncOperation();
```

Comment:
Is there a reason why the initalization is not completely async ? Is this operation fast ?

Reply:
As the other platforms don't have asynchronous retrieval of the sensor, I didn't consider it originally. It is fast, but I could wrap the initialization in a `Task.Run` if that would be more appropriate.

Reply:
Done :-) .